### ISSUE TITLE
Swallows helpful error if there is malformed package.json

### ISSUE THREAD

**Issue by chadhietala** [notion-cli/notion#280]
If you have malformed JSON in your package.json the error gets swallowed by Notion.

**Steps To Reproduce**

1. Edit package.json and leave a dangling comma
2. `yarn`
3. See the following error:

```
❯ yarn
Notion is still a pre-alpha project, so we expect to run into some bugs,
but we'd love to hear about them so we can fix them!

Please feel free to reach out to us at @notionjs on Twitter or file an issue at:

    https://github.com/notion-cli/notion/issues
```

**Expectation**

Malformed package.json should result in throwing the package managers error:

```
error An unexpected error occurred: "/Users/chietala/Code/some-package/package.json: Unexpected token ] in JSON at position 426".
```

**Comment by charlespierce** [comment]
Agreed! Working on improving the error messages to be helpful and point you to the right place is the one of the priorities right now. I have PR #249 open to lay the technical groundwork for improving the error messages, then we'll be creating a quest issue to tackle all of the outstanding "An unknown error occurred" issues.

See also #199 for a related issue about malformed package.json (in that case because of a merge conflict marker)

**Comment by charlespierce** [comment]
For reference (from @thoov in #315): Yarn outputs the following with a malformed `package.json`:

```
error An unexpected error occurred: "/path/to/project/package.json: Unexpected token : in JSON at position 8".
info If you think this is a bug, please open a bug report with the information provided in "/path/to/project/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

**Comment by chriskrycho** [comment]
I also observe here, and relevant to #362, that there's not actually any way to get at the verbose output from Notion when running a shimmed command. A shimmed `yarn` will fail like this with a malformed `package.json`:

```sh
$ yarn
Notion error: Could not parse project manifest
at /Users/chris/dev/test/package.json

Please ensure that the file is correctly formatted.
Error details written to: notion-error-2019-05-01_14_08_52.012.log
```

This is an improvement, for sure, and #395 helps as well, but this also suggests to me a related point (which likely deserves its own issue): for shimmed commands in general we need a way for users to surface the verbose error information from the shimmed command. 🤔

**Comment by charlespierce** [comment]
There actually is an (undocumented) way to get verbose errors when running a shimmed command: Setting the `NOTION_DEV` environment variable. It's likely worthwhile to change the name to e.g. `NOTION_VERBOSE` that and officially document it as a way of exposing "verbose mode" when working with shims.
